### ISSUE TITLE
Bug: [no-extraneous-import] should allow type imports if @types package is installed

### ISSUE THREAD

**Issue by danielrentz** [eslint-community/eslint-plugin-n#486]
### Environment

Node version: 22.19.0
npm version: 9.8.1
ESLint version: 9.36
eslint-plugin-n version: 17.23.1
Operating System: Win11

### What rule do you want to report?

no-extraneous-import

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/node-h8zzzzog

### What did you expect to happen?

In TS code, `no-extraneous-import` should allow type imports from module `X` if `@types/X` package is installed although `X` is not.

Example:

```sh
> npm install @types/picomatch
```

```ts
import type { Glob } from "picomatch";
                       // ^^^^^^^^^^^ "picomatch" is extraneous.

export type { Glob } from "picomatch";
                       // ^^^^^^^^^^^ "picomatch" is extraneous.
```

### Participation

- [ ] I am willing to submit a pull request for this issue.
